cmd/launcher/main.go

```diff
@@ -88,13 +88,33 @@ func runMain() int {
 	// fork-bombing itself. This is an ENV, because there's no
 	// good way to pass it through the flags.
 	if !env.Bool("LAUNCHER_SKIP_UPDATES", false) && !inBuildDir {
-		if err := runNewerLauncherIfAvailable(ctx, systemSlogger.Logger); err != nil {
-			systemSlogger.Log(ctx, slog.LevelInfo,
-				"could not run newer version of launcher",
+		lastestLauncherPath, err := latestLauncherPath(ctx, systemSlogger.Logger)
```

Comment:
Could we throw a quick comment in here saying that we do this rather than calling `runNewerLauncherIfAvailable` so that we can distinguish between "exited with error" and "exited without error"? I could see us forgetting why we made this change and trying to tidy up/refactor, and accidentally consolidating back into `runNewerLauncherIfAvailable` again.